src/track.py

```diff
@@ -72,16 +96,33 @@ def eval_seq(opt, dataloader, data_type, result_filename, save_dir=None, show_im
         timer.toc()
         # save results
         results.append((frame_id + 1, online_tlwhs, online_ids))
-        if show_image or save_dir is not None:
+        if show_image is not None:
             online_im = vis.plot_tracking(img0, online_tlwhs, online_ids, frame_id=frame_id,
                                           fps=1. / timer.average_time)
         if show_image:
             cv2.imshow('online_im', online_im)
-        if save_dir is not None:
-            cv2.imwrite(os.path.join(save_dir, '{:05d}.jpg'.format(frame_id)), online_im)
-        frame_id += 1
+        
+        # A neat trick I learned from PyImageSearch.com / Adrian Rosebrock's book 
+        if writerProcess is None:
+            writeVideo = Value('i', 1)
+            if W is None or H is None:
+                (H, W) = online_im.shape[:2]
+            frameQueue = Queue()
+            writerProcess = Process(target=write_video, args=(
+                result_root.replace(".mp4",".avi"), writeVideo, frameQueue, W, H, frame_rate))
+            writerProcess.start()
+        
+        if writerProcess is not None:
+            frameQueue.put(online_im)
+    frame_id += 1
     # save results
     write_results(result_filename, results, data_type)
+    if writerProcess is not None:
+        writeVideo.value = 0
+        writerProcess.join()
+        ff = FFmpeg(inputs={result_root.replace(".mp4",".avi"): None}, outputs={result_root: '-vcodec h264 -acodec mp2'})
```

Comment:
Maybe it's better to write here like this?
Outputs under inputs.

        ff = FFmpeg(inputs={result_root.replace(".mp4", ".avi"): None},
                            outputs={result_root: '-vcodec h264 -acodec mp2'})

I think this will increase the readability of the code